compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/PackageLoader.java

```diff
@@ -283,14 +290,14 @@ public BLangPackage loadEntryPackage(PackageID pkgId, PackageID enclPackageId, b
         return packageNode;
     }
 
-    public BLangPackage loadPackage(PackageID pkgId, PackageID enclPackageId, PackageRepository packageRepo) {
+    private BLangPackage loadPackage(PackageID pkgId) {
         // TODO Remove this method()
         BLangPackage bLangPackage = packageCache.get(pkgId);
         if (bLangPackage != null) {
             return bLangPackage;
         }
-
-        BLangPackage packageNode = loadPackageFromEntity(pkgId, loadPackageEntity(pkgId, enclPackageId));
+    
+        BLangPackage packageNode = loadPackageFromEntity(pkgId, loadPackageEntity(pkgId, null, null));
```

Comment:
can we introduce a new method instead passing null, can be confusing to read.

Reply:
Created a separate method, Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/10442/commits/dffb67936effdf7af12306849aa3c093cefc19d7